specifyweb/stored_queries/query_construct.py

```diff
@@ -4,17 +4,12 @@
 from sqlalchemy import orm, sql
 
 from specifyweb.specify.models import datamodel
+from specifyweb.specify.tree_utils import get_treedef
 
-from . import models
+from specifyweb.stored_queries import models
 
 logger = logging.getLogger(__name__)
 
-
-def get_treedef(collection, tree_name):
-    return (collection.discipline.division.institution.storagetreedef
-            if tree_name == 'Storage' else
-            getattr(collection.discipline, tree_name.lower() + "treedef"))
-
```

Comment:
I don't think what you have will work. IIRC, we were going to support queries to ALL trees, correct? So if you get "species", it'll return species in Mineral and Fossil tree. This doesn't do that...

Previously, I mentioned adding that COT in tree stringid. We don't technically need that anymore (if we're searching for all trees), but what you have wouldn't work. For instance, try searching for a rank = 'blahh'. you won't get the results for _all_ the trees. 

I'll recommend looking at how it constructs tree joins to figure out why. You even have a debugger!

Reply:
You _really_ need to lookup all the treedefids, and then get the most depth here (https://github.com/specify/specify7/blob/dff3564be2f8e4b5a5a60ba761f2cbac995801e8/specifyweb/stored_queries/query_construct.py#L43), and then look at ALL the treedefitemids (https://github.com/specify/specify7/blob/dff3564be2f8e4b5a5a60ba761f2cbac995801e8/specifyweb/stored_queries/query_construct.py#L56) that match a particular rank

Basically, that should be an OR (or an IN, doesn't matter. So, use OR -- more efficient)

Reply:
> we were going to support queries to ALL trees, correct? So if you get "species", it'll return species in Mineral and Fossil tree. This doesn't do that...

correct

Reply:
👍, the issue I mentioned should be verified and resolved then. I can't set it up on testpanel due to this [commit](https://github.com/specify/specify7/actions/runs/10115757864/job/27977226991) presumably